### ISSUE TITLE
sagews -- Python comments are being stripped when parsing % blocks

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/smc#978]
To replicate:

1. In a Sage worksheet, define a mode
```
def f(s):
    print "s='%s'"%s
```
2. Use it:
```
%f
123
# foo
456
```
3. Expected output should have foo in it, but does not.

See https://cloud.sagemath.com/projects/4a5f0542-5873-4eed-a85c-a18c706e8bcd/files/support/2016-10-01-072137-parsing-comments.sagews

**Comment by haraldschilly** [comment]
Did the tests pass before you pushed this?! I've tried twice, and both time the `issue819` fails with

```
>               assert mesg['stdout'] == output
E               assert 'should not execute 2' == '22\n'
E                 - should not execute 2
E                 + 22
```

**Comment by williamstein** [comment]
I didn't run the tests.

**Comment by haraldschilly** [comment]
I'm pretty sure this breaks #819

**Comment by williamstein** [comment]
Now 981 -- https://github.com/sagemathinc/smc/issues/981

In any case, the fix that this includes is imho much more important than #819.

**Comment by williamstein** [comment]
In retrospect, my fix here was basically "revert #819", and the fix for 819 was definitely wrong.
